### PR TITLE
I-#269: Updated accept donation api to also handle ignore response

### LINKED ISSUE #269: Create API to add ignore status on Ignore
Update `donations/accept` to add a status field. for status ignore only change the notification status. and for accept continue with. 
also chang the end point to `donations/response`.

### PR DESCRIPTION
**Pull request by mahinuzzaman**
Fixes #269 
- Updated accept donation api to also handle ignore response
- Fixed Notification fields missing bugs
- Updated notification DTO and types
- Moved accept API to `donation/responses` to handle both accept and ignore status

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [x] New feature
- [x] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: core/services/aws/bloodDonation/acceptDonationRequest.ts [hunk 1]
```diff
@@ -45,88 +45,189 @@ async function acceptDonationRequestLambda(
...
-    const donationPost = await bloodDonationService.getDonationRequest(
-      event.seekerId,
-      event.requestPostId,
-      event.createdAt,
-      new BloodDonationDynamoDbOperations<DonationDTO, DonationFields, BloodDonationModel>(
-        new BloodDonationModel()
+
+    if (status === AcceptDonationStatus.ACCEPTED && acceptanceRecord === null) {
+      await acceptBloodDonationRequest(donorId, seekerId, requestPostId, createdAt, status)
+    } else if (status === AcceptDonationStatus.IGNORED && acceptanceRecord !== null) {
```
Comment (sakibstark11) [inline core/services/aws/bloodDonation/acceptDonationRequest.ts:66]: else if without an else?

File: core/services/aws/bloodDonation/acceptDonationRequest.ts [hunk 2]
```diff
@@ -45,88 +45,189 @@ async function acceptDonationRequestLambda(
...
+      { message: 'Donation request accepted successfully.' },
+      HTTP_CODES.OK
+    )
+  } catch (error) {
+    const errorMessage = error instanceof Error ? error.message : 'An unknown error occurred'
+    return generateApiGatewayResponse(`Error: ${errorMessage}`, HTTP_CODES.ERROR)
+  }
+}
+
+export default acceptDonationRequestLambda
```
Comment (sakibstark11) [inline core/services/aws/bloodDonation/acceptDonationRequest.ts:100]: is this necessary? it goes outside of our settings

**Review by sakibstark11 — DISMISSED** [review]

File: core/services/aws/bloodDonation/acceptDonationRequest.ts (hunk 1 above)
Comment (mahinuzzaman) [reply core/services/aws/bloodDonation/acceptDonationRequest.ts]: updated in f55b70d04d0d650f746b701e4740aa308ba5ee77

File: core/services/aws/bloodDonation/acceptDonationRequest.ts [hunk 3]
```diff
@@ -27,106 +27,216 @@ import {
...
+      >(new AcceptDonationRequestModel())
     )
-    if (donationPost.status !== DonationStatus.PENDING) {
-      throw new Error('Donation request is no longer available for acceptance.')
+    if (acceptanceRecord !== null && acceptanceRecord.status === AcceptDonationStatus.COMPLETED) {
+      throw new Error('You already donated.')
+    }
+
+    if ((status === AcceptDonationStatus.ACCEPTED && acceptanceRecord !== null)) {
+      throw new Error('You already accepted the donation')
```
Comment (sakibstark11) [inline core/services/aws/bloodDonation/acceptDonationRequest.ts:68]: i am thinking if we should throw errors here? idempotency dictates that the same request should have the same output.

Comment (mahinuzzaman) [reply core/services/aws/bloodDonation/acceptDonationRequest.ts]: Updated in d1a75e62dc44681507dbd3b131846c6a53ab9363

**Review by sakibstark11 — APPROVED** [review]

File: core/services/aws/bloodDonation/acceptDonationRequest.ts (hunk 3 above)
Comment (sakibstark11) [reply core/services/aws/bloodDonation/acceptDonationRequest.ts]: i think the message should be different though? it should still say, he this alrady exists but we just uswe 200

Comment (mahinuzzaman) [reply core/services/aws/bloodDonation/acceptDonationRequest.ts]: got it. will update in next pr

**Pull request merged**